### PR TITLE
Add typing for plugins/__init__.py

### PR DESCRIPTION
**Pull request by ukulekek**

### CODE REVIEW

File: tmt/plugins/__init__.py [hunk 1]
```diff
@@ -14,11 +15,11 @@
 log = fmf.utils.Logging('tmt').logger
 
 
-def explore():
+def explore() -> None:
     """ Explore all available plugins """
 
     # Check all tmt steps for native plugins
-    root = os.path.dirname(os.path.realpath(tmt.__file__))
+    root: str = os.path.dirname(os.path.realpath(tmt.__file__))
```
Comment (happz) [inline tmt/plugins/__init__.py:22]: What was the error message that made you add an explicit `str` annotation? I would expect the type of `root` to be easy to infer, thanks to `os.path.dirname` being a standard library function with a known return type value (`AnyStr`, or `str` in Python 3).

Comment (ukulekek) [reply tmt/plugins/__init__.py]: No errors, I just overtyped it a little :) Reversing the lone, thanks

File: tmt/plugins/__init__.py [hunk 2]
```diff
@@ -6,6 +6,7 @@
 import os
 import pkgutil
 import sys
+from typing import Generator, List, Optional
```
Comment (thrix) [inline tmt/plugins/__init__.py:9]: @ukulekek List is not needed afaict

```suggestion
from typing import Generator, Optional
```

**Review by happz — APPROVED** [review]

**Review by thrix — APPROVED** [review]
LGTM

**Pull request merged**